libs/ledger/src/chaincode/chain_code_cache.cpp

```diff
@@ -30,30 +30,29 @@
 namespace fetch {
 namespace ledger {
 
-ChainCodeCache::ContractPtr ChainCodeCache::Lookup(Identifier const &contract_id,
-                                                   StorageInterface &storage)
+ChainCodeCache::ContractPtr ChainCodeCache::Lookup(ConstByteArray const &contract_id,
+                                                   StorageInterface &    storage)
 {
   // attempt to locate the contract in the cache
   ContractPtr contract = FindInCache(contract_id);
 
   // if this fails create the contract
   if (!contract)
   {
-    if (contract_id.type() == Identifier::Type::SMART_OR_SYNERGETIC_CONTRACT)
+    chain::Address address;
+    if (chain::Address::Parse(contract_id, address))
     {
-      auto const contract_digest = contract_id.qualifier().FromHex();
-      contract                   = CreateSmartContract<SmartContract>(contract_digest, storage);
+      contract = CreateSmartContract<SmartContract>(address.display(), storage);
```

Comment:
pass address.address() instead of display, which returns a base58 string. Let's keep things as raw bytes as much as possible.

Reply:
As discussed, pass in Address, call .display() inside CreateSmartContract